### ISSUE TITLE
"CondaEnvironment" is too slow in windows

### ISSUE THREAD

**Issue by guolinke** [Microsoft/vsts-tasks#8456]
refer to log: https://lightgbm-ci.visualstudio.com/lightgbm-ci/_build/results?buildId=256&view=logs

the task "CondaEnvironment" always costs about 5 minutes in windows, while it is only about 1 minutes in Linux, and 30 seconds in macos.

Any idea to speed up it ?

**Comment by guolinke** [comment]
Thanks! BTW,  does ubuntu vm have the similar “d” drive to speed up?

**Comment by brcrista** [comment]
I don't think so.

**Comment by guolinke** [comment]
@brcrista 
Is that possible to build a customed windows docker, which pre-installs these packages, by ourself?

**Comment by brcrista** [comment]
Yes, you can self-host agents: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/v2-windows?view=vsts

**Comment by vtbassmatt** [comment]
@guolinke if you've got a container you want to use, we can run that as a [container job](https://docs.microsoft.com/en-us/azure/devops/pipelines/process/container-phases?view=vsts&tabs=yaml). For Windows containers, you'll need to set:
```yaml
pool:
  vmImage: win1803
```
